src/main/java/io/github/railroad/utility/localization/L18n.java

```diff
@@ -0,0 +1,69 @@
+package io.github.railroad.utility.localization;
+
+import io.github.railroad.Railroad;
+import io.github.railroad.utility.ConfigHandler;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import static io.github.railroad.Railroad.LOGGER;
+
+public class L18n {
+    private static final Properties LANG_CACHE = new Properties();
+    private static final ObjectProperty<Languages> CURRENT_LANG = new SimpleObjectProperty<>();
+
+    private L18n() {
+    }
+
+    public static void setLanguage(Languages language) {
+        //Updates the config and calls loadLanguage to update the cache and CURRENT_LANG
+        LOGGER.debug("Setting language to {}", language);
+
+        var updated = ConfigHandler.getConfigJson();
+        updated.get("settings").getAsJsonObject().addProperty("language", language.toString());
+
+        ConfigHandler.updateConfig();
+        loadLanguage();
+    }
+
+    public static Languages getCurrentLanguage() {
+        return CURRENT_LANG.getValue();
+    }
+
+    public static ObjectProperty<Languages> currentLanguageProperty() {
+        return CURRENT_LANG;
+    }
+
+    public static void loadLanguage() {
+        //Loads the language into cache and sets the CURRENT_LANG
+        LOGGER.info("Loading language file");
+        Languages newLang = Languages.valueOf(ConfigHandler.getConfigJson().get("settings").getAsJsonObject().get("language").getAsString());
+
+        try {
+            InputStream props = Railroad.getResourceAsStream("lang/" + newLang + ".properties");
```

Comment:
Before we go too far with this, its probably worth having them be `.lang` extensions, not `.properties`